### DISCUSSION TITLE
How tagger.predict works?

### DISCUSSION THREAD

**Discussion by dglopes** [explosion/spaCy#13193] [Help: Installation] [answered]
I was looking for some way to measure the POS tagging confidence and found this answer:
https://stackoverflow.com/questions/65218606/is-it-possible-to-find-uncertainties-of-spacy-pos-tags
But after some research I didnt figured out how the tagger.predict method works or measure the tag probability/confidence.

**Comment by adrianeboyd** [comment, marked as answer]
You can get the raw scores from the tagger's internal model:

```
nlp = spacy.load("en_core_web_sm")
doc = nlp("This is a sentence.")
nlp.get_pipe("tagger").model.predict([doc])
```

The scores for each token correspond to the labels in `nlp.get_pipe("tagger").labels`.

There's a bit of cython, but the pair of `predict` / `set_annotations` methods should help you understand the internals:

https://github.com/explosion/spaCy/blob/7df328fbfe42c1415f5e324686f7c4dcc0c093cb/spacy/pipeline/tagger.pyx#L124-L172

**Reply by dglopes** [reply]
So it returns the accuracy of the model for that token labels array based on its attempt to classify/predict the corret label?
ref: https://spacy.io/api/pipe

**Reply by adrianeboyd** [reply]
I'm not sure what you mean? It returns the model's score for each possible tag for each token in the doc.

**Reply by dglopes** [reply]
the score is given by model.predict, but what calculation occurs in the predict method? I imagine it is the model's accuracy measure calculated by training...but I'm not sure. Will the accuracy be measured based on your attempt x gold-standard?

**Reply by dglopes** [reply]
Ah, its ok! I found the explanation here:
https://spacy.io/usage/training

:)
